config/application.rb

```diff
@@ -298,7 +298,10 @@ def cache_store_config
 
     config.cms_files_path = ':url_root/:date_partition/:basename-:random_secret.:extension'
 
-    require 'three_scale/deprecation'
+    # Add a custom deprecator, silenced in test and production
+    Rails.application.deprecators[:threescale] = ActiveSupport::Deprecation.new('future version', '3scale')
+    Rails.application.deprecators[:threescale].silenced = %w[test production].include?(Rails.env)
```

Comment:
Couldn't place this code to the `config/initializers/deprecation.rb`, because it get loaded **after** some models that use `Rails.application.deprecators[:threescale]` are getting loaded.

Also, with this refactoring the classes `ThreeScale::Deprecation` and `ThreeScale::Deprecation::Deprecator` are not needed anymore.

Reply:
Yes, I believe I tried to keep the existing functionality.
Indeed, this is just some messages like:

```
>> Account.first.domain
DEPRECATION WARNING: domain is deprecated and will be removed from 3scale future version (use #internal_domain or #external_domain) 

>> Metric.first.domain
DEPRECATION WARNING: service is deprecated and will be removed from 3scale future version (refer to #owner)
```

We'll see those in development environment, no need to see in production, and probably in test either. Unless someone is doing a refactoring, and wants to run a test suite to verify everything is fixed - in this case it can be enabled at this point.